src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AlignmentInterval.java

```diff
@@ -5,43 +5,158 @@
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.seqdoop.hadoop_bam.SAMFormat;
 
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 
-import static org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection.MISSING_NM;
-
 /**
  * Each assembled contig should have at least one such accompanying structure, or 0 when it is unmapped.
  */
 @DefaultSerializer(AlignmentInterval.Serializer.class)
 public final class AlignmentInterval {
 
+    /**
+     * Indicates that no NM (number of mismatches) annotation is present.
+     */
+    public static final int NO_NM = -1;
+
+    /**
+     * Indicates that no AS (alignment score) annotation is present.
+     */
+    public static final int NO_AS = -1;
+
     public final SimpleInterval referenceSpan;
     public final int startInAssembledContig;   // 1-based, inclusive
     public final int endInAssembledContig;     // 1-based, inclusive
 
     public final Cigar cigarAlong5to3DirectionOfContig;
 
     public final boolean forwardStrand;
+
+    /**
+     * Mapping quality, {@link SAMRecord#NO_MAPPING_QUALITY} if unspecified.
+     */
     public final int mapQual;
+
+    /**
+     * Number of mismatches, {@link #NO_NM} if unknown or unspecified.
+     */
     public final int mismatches;
+
+    /**
+     * Alignment score, {@link #NO_AS} if unknown or unspecified.
+     */
     public final int alnScore;
 
     // if any of the following boolean fields are true, fields "mapQual", "mismatches", "alnScore" should be viewed
     // with care as they were simply copied from the original alignment (not for "mismatches"),
     // which after the split are wrong (we didn't recompute them because that would require expensive SW re-alignment)
+
     public final boolean isFromSplitGapAlignment;
     public final boolean hasUndergoneOverlapRemoval;
 
+    /**
+     * Compose an alignment interval instance from a SAM supplementary alignment formatted string.
+     * <p>
+     *     The input string format is:
+     *     <pre>
+     *         chr-name,start,strand,cigar,mq,nm,as
+     *     </pre>
+     *     where mq (mapping-quality), nm (number of mismatches) and as (alignment score) might be absent. The strand
+     *     is symbolized as '+' for the forward-strand and '-' for the reverse-strand.
+     *     <p>Examples:</p>
+     *     <pre>
+     *         chr10,1241241,+,10S1313M45I14M100H,30,5,66
+     *         chr10,1241241,+,10S1313M45I14M100H,30,5
+     *         chr10,1241241,+,10S1313M45I14M100H,30
+     *         chr10,1241241,+,10S1313M45I14M100H
+     *     </pre>
+     * </p>
+     * @param str the input string.
+     * @throws IllegalArgumentException if {@code str} is {@code null} or it does not look like a valid
+     *   SA string.
+     */
+    public AlignmentInterval(final String str) {
```

Comment:
Not sure about this use of  [at]VisibleForTesting.... my understanding is that it should be use on public or package methods that are so only for the sake of make possible to be directly accessible by testing code. Is a way to say "The only reason this is visible is for the sake of testing". 

If i'm correct the you should be marking public methods with this annotation that are mean to be public as that is misleading. 

This is the javadoc:

package com.google.common.annotations;
```

/**
 * Annotates a program element that exists, or is more widely visible than
 * otherwise necessary, only for use in test code.
 *
 * @author Johannes Henkel
 */
@GwtCompatible
public @interface VisibleForTesting {
}

```